tests/test_cr_history_indexing.py

```diff
@@ -0,0 +1,156 @@
+        super().__init__(
+            name, owner, description, beforeTests, afterTests, tags)
+
+
+def isEqualComponentRun(crOne, crTwo):
+    if crOne.start_timestamp == crTwo.start_timestamp \
+            and crOne.end_timestamp == crTwo.end_timestamp:
+        return True
+    return False
+
+
+class TestComponentRunHistory(unittest.TestCase):
+    def setUp(self):
+        self.store = Store("test")
+
+        # initialize four component run for TestHistoryComponent
+        self.store.create_component("mock_component", "", "")
+        num_runs = 4
+        for i in range(1, num_runs + 1):
+            inp = self.store.get_io_pointer("inp")
+            out = self.store.get_io_pointer("out")
+            cr = self.store.initialize_empty_component_run("mock_component")
+            cr.set_start_timestamp()
+            cr.set_end_timestamp()
+            cr.add_input(inp)
+            cr.add_output(out)
+            self.store.set_dependencies_from_inputs(cr)
+            self.store.commit_component_run(cr)
+
+        self.historyLength = self.store.get_component_runs_count(
+            "mock_component")
+        self.firstComponentRun = self.store.get_component_runs_by_index(
+            "mock_component", 0, 1)[0]
+        self.lastComponentRun = self.store.get_component_runs_by_index(
+            "mock_component", self.historyLength - 1, self.historyLength)[0]
+        self.secondAndThirdComponentRun = \
+            self.store.get_component_runs_by_index(
+                "mock_component", 1, 3)
+
+    def testPstPstIndex(self):
+
+        # case 1: (0, 0) return zero componentRun
+        resCrList = self.store.get_component_runs_by_index(
+            "mock_component", 0, 0)
+        self.assertEqual(len(resCrList), 0)
+
+        # case 2: (0, 1) return the first componentRun
+        resCrList = self.store.get_component_runs_by_index(
+            "mock_component", 0, 1)
+        self.assertEqual(len(resCrList), 1)
+        self.assertTrue(isEqualComponentRun(
+            resCrList[0], self.firstComponentRun))
+
+        # case 3: (len(componentRun) - 1, len(componentRun))
+        # return the last componentRun
+        resCrList = self.store.get_component_runs_by_index(
+            "mock_component", self.historyLength - 1, self.historyLength)
+        self.assertEqual(len(resCrList), 1)
+        self.assertTrue(isEqualComponentRun(
+            resCrList[0], self.lastComponentRun))
+
+        # case 4: (1, 3) return the second and third componentRun
+        # given len(componentRun) >=3)
+        resCrList = self.store.get_component_runs_by_index(
+            "mock_component", 1, 3)
+        self.assertEqual(len(resCrList), 2)
+        for idx, cr in enumerate(resCrList):
+            if idx >= len(self.secondAndThirdComponentRun):
+                break
+            else:
+                self.assertTrue(isEqualComponentRun(
+                    cr, self.secondAndThirdComponentRun[idx]))
+
+    def testPstNgtIndex(self):
```

Comment:
What does Pst and Ngt mean? Feel free to make the function names long or write a comment description

Reply:
Write a comment but in short, testPstPstIndex means test all possibilities with get_component_runs_by_index(positive idx, positive idx).

Reply:
move the comment below the function name, like:

```
def testPstNgtIndex(self):
  """
 Test all possible queries:
        get_component_runs_by_index(positive idx, negative idx)
  """
```